addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2035,6 +2055,38 @@ const SnippetOptionWidget = Widget.extend({
                 });
             });
         }});
+
+        // Check linked widgets: force their value and simulate a notification
+        const triggerWidgetsNames = ev.data.triggerWidgetsNames;
+        const triggerWidgetsValues = ev.data.triggerWidgetsValues;
+        for (let i = 0, l = triggerWidgetsNames.length; i < l; i++) {
+            const widgetName = triggerWidgetsNames[i];
+            const widgetValue = triggerWidgetsValues[i];
+
+            let linkedWidget = null;
+            this.trigger_up('user_value_widget_request', {
+                name: widgetName,
+                onSuccess: _widget => linkedWidget = _widget,
+            });
+            if (linkedWidget) {
+                if (widgetValue !== undefined) {
+                    // FIXME right now only make this work supposing it is a
+                    // colorpicker widget with big big hacks, this should be
+                    // improved a lot
+                    const normValue = this._normalizeWidgetValue(widgetValue);
+                    if (previewMode === true) {
+                        linkedWidget._previewColor = normValue;
+                    } else if (previewMode === false) {
+                        linkedWidget._previewColor = false;
+                        linkedWidget._value = normValue;
+                    } else {
+                        linkedWidget._previewColor = false;
+                    }
+                }
+
+                linkedWidget.notifyValueChange(previewMode, true);
```

Comment:
Possible solution: make UserValueWidgets implement a method 
```
select(value, previewMode)
```
instead of having them call onUserValueXXX, to decouple the logic of extracting the data from the original context (eg color events' data.color), and handling the logic for previewMode. This way you can  call `linkedWidget.select(widgetValue, true)` here, while in the colorpicker you would do for example:

```javascript
select: function (value, previewMode) {
    this.previewColor = previewMode === true ? value : false;
    if (!previewMode) {
        this._value = value;
    }
    this._onUserValueUpdate(value, previewMode); // Ideally we'd want this to happen automagically after calling select, but I don't see how, maybe use a decorator from the super class.
}

_onColorHovered: function (ev) {
    this.select(ev.data.color, true);
}

_onColorPicked: function (ev) {
    this.select(ev.data.color, false);
}

_onColorLeft: function (ev) {
    this.select(ev.data.color, 'reset');
}

Reply:
Yes and no. I thought about extending `setValue` with an additional `previewMode` parameter (which is what you suggest) but it should handle the fact a value can be linked to a given method (current second parameter of the `setValue` method). To think about.